lib/mods/replace-get-set.js

```diff
@@ -25,25 +25,70 @@ function isDataProperty(type, propertyName) {
 }
 
 /**
- * @param {Node} node - the current ts-morph AST Node
+ * @param {Node} node - the ts-morph AST Node to evaluate
  *
- * @return {boolean} whether the node is a global get
+ * @return {boolean} whether the node is a global get/set
  */
-function isGlobalGet(node) {
+function isGlobalPropertyAccess(node) {
   return nodeTypes.isPropertyAccessExpression(node) &&
     !nodeTypes.isCallExpression(node.getParent()) &&
     isGlobal(node.getExpression());
 }
 
 /**
- * @param {Node} node - the current ts-morph AST Node
+ * @param {Node} node - the ts-morph AST Node to evaluate
  *
  * @return {boolean} whether the node is a global set
  */
-function isGlobalSet(node) {
+function isSet(node) {
   return nodeTypes.isBinaryExpression(node) &&
-    node.getOperatorToken().getText() === '=' &&
-    isGlobalGet(node.getLeft());
+    node.getOperatorToken().getText() === '=';
+}
+
+/**
+ * Checks if node in question is a non-property access window
+ * identifier, by eliminating other possible types of globals
+ * the identifier could be
+ *
+ * @param {Node} node - the ts-morph AST Node to evaluate
+ *
+ * @return {boolean} whether the node is a window global
+ */
+function isWindowGlobal(node) {
+  // filter non-globals
+  const symbol = node.getSymbol();
+  const declaration = symbol && symbol.getDeclarations()[0];
+  if (!declaration ||
+    !declaration.getSourceFile().getFilePath()
+        .includes('/node_modules/typescript')) {
+    return false;
+  }
+
+  // filter constructors - e.g. Document
+  const baseConstructSignature = node.getType().getConstructSignatures()[0];
+  if (baseConstructSignature) {
+    return false;
+  }
+
+  // filter namespaces - e.g. CSS
+  if (node.getType().getText() === node.getText()) {
+    return false;
+  }
+
+  // filter data properties - e.g. array.length
+  const parent = node.getParent();
+  const propertyName = node.getText();
+  let baseName;
+
+  if (nodeTypes.isPropertyAccessExpression(parent)) {
```

Comment:
Why do we need this check given the check on line 145? What test fails if you remove it?

Reply:
It fails with an error replacing tree in get-set and property-access, the culprit being an array.length which seems to mess things up.